server.js

```diff
@@ -7,23 +7,68 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
-// Start the server
+app.get('/thoughts', async (req, res) => {
+  try {
+    const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+    res.json(thoughts)
+  }
+  catch {
+    res.status(404).json({ error: "Can not find thoughts" })
+  }
+})
+
+app.post('/thoughts', async (req, res) => {
+  try {
+    const { message } = req.body
+    const thought = new Thought({ message })
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  }
+  catch (err) {
+    res.status(400).json({ message: 'Could not save thought', error: err.errors })
+  }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params
+
+  try {
+    await Thought.updateOne({ _id: thoughtId }, { $inc: { hearts: 1 } })
+    const likeThought = await Thought.findOne({ _id: thoughtId })
```

Comment:
This could be done in one line using `findOneAndUpdate()`:
`const likedThought = await Thought.findOneAndUpdate({ _id: req.params.id }, { $inc: { hearts: 1 } }, { new: true })`